src/components/table/table-modern.style.js

```diff
@@ -0,0 +1,71 @@
+import { css } from 'styled-components';
+import StyledTableCell from './table-cell/table-cell.style';
+import StyledTableHeader from './table-header/table-header.style';
+
+function convertTableType(type) {
+  if (['primary', 'secondary', 'tertiary'].includes(type)) return type;
+  if (type === 'transparent') return 'tertiary';
+  if (type === 'light') return 'secondary';
+  return 'primary';
+}
+
+function applyModernInternalStyling({ tableType, theme }) {
+  const type = convertTableType(tableType);
+  return css`
+    background-color: ${type === 'tertiary' ? 'transparent' : theme.table.primary};
+    border: 1px solid ${type === 'tertiary' ? 'transparent' : theme.table.primary};
+  `;
+}
+
+function applyModernTableStyling({
+  tableType, theme, isZebra, size
+}) {
+  const type = convertTableType(tableType);
+  const { table, colors } = theme;
+
+  return css`
+    background-color: ${type === 'tertiary' ? 'transparent' : theme.colors.white};
+    && .carbon-table-row {
```

Comment:
&&

Reply:
It raises the priority of the style to allow it to override what's already defined